### PR TITLE
update the translation of `Object.seal()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description
update the translation of `Object.seal()`

### Motivation
update the translation of `Object.seal()`

### Additional details

none

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 1]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
 
 {{JSRef}}
 
-**`Object.seal()`** 方法封闭一个对象，阻止添加新属性并将所有现有属性标记为不可配置。当前属性的值只要原来是可写的就可以改变。
+**`Object.seal()`** 静态方法*密封*一个对象。密封一个对象[阻止其扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)并且使得现有属性不可配置。密封对象有一组固定的属性：不能添加新属性、不能删除现有属性、不能更改其可枚举性和可配置性、不能重新分配其原型。只要现有属性的值是可写的，它们仍然可以更改。`seal()` 返回传入的同一对象。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:8]:
```suggestion
**`Object.seal()`** 静态方法*密封*一个对象。密封一个对象会[阻止其扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)并且使得现有属性不可配置。密封对象有一组固定的属性：不能添加新属性、不能删除现有属性或更改其可枚举性和可配置性、不能重新分配其原型。只要现有属性的值是可写的，它们仍然可以更改。`seal()` 返回传入的同一对象。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 2]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
+  - : 要密封的对象。
 
 ### 返回值
 
 被密封的对象。
 
 ## 描述
 
-通常，一个对象是{{jsxref("Object.isExtensible()", "可扩展的", "", 1)}}（可以添加新的属性）。密封一个对象会让这个对象变的不能添加新属性，且所有已有属性会变的不可配置。属性不可配置的效果就是属性变的不可删除，以及一个数据属性不能被重新定义成为访问器属性，或者反之。但属性的值仍然可以修改。尝试删除一个密封对象的属性或者将某个密封对象的属性从数据属性转换成访问器属性，结果会静默失败或抛出{{jsxref("TypeError")}}（在{{jsxref("Strict_mode", "严格模式", "", 1)}} 中最常见的，但不唯一）。
+密封一个对象相当于[阻止其扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，然后将现有的[属性描述符](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#描述)更改为 `configurable: false`。这有使对象上的属性固定的效果。使其所有属性不可配置且可以防止它们从数据属性转换成访问器属性，反之亦然，但它不会阻止数据属性值的更改。尝试删除或者向密封对象添加属性、或者将数据属性转换成访问器属性、或者反过来，都将失败，无论是静默还是抛出 {{jsxref("TypeError")}}（最常见的，尽管不是唯一，当处于 {{jsxref("Strict_mode", "严格模式", "", 1)}} 代码中）。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:29]:
```suggestion
密封一个对象等价于[阻止其扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，然后将现有的[属性描述符](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#描述)更改为 `configurable: false`。这会使对象上的属性集固定。令对象的所有属性不可配置且可以防止它们从数据属性转换成访问器属性，反之亦然，但它不会防止数据属性的值被更改。尝试删除或者向密封对象添加属性，或者将数据属性和访问器属性互相转换，都将失败，可能会静默失败，也可能会抛出 {{jsxref("TypeError")}}（最常见的是在{{jsxref("Strict_mode", "严格模式", "", 1)}}代码中，但不仅限于此）。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 3]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
 
-不会影响从原型链上继承的属性。但 [`Object.prototype.__proto__`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/proto) ( {{deprecated_inline}} ) 属性的值也会不能修改。
+原型链保持原样。然而，由于[阻止扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，`[[Prototype]]` 不能重新分配。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:31]:
```suggestion
原型链保持原样。然而，由于[扩展被阻止](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，`[[Prototype]]` 不能重新分配。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 4]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
+Object.defineProperty(obj, "foo", {
+  value: "eit",
+}); // 更改现有属性值
 ```
 
-## 注意
+### 非对象参数
 
-在 ES5 中，如果这个方法的参数不是一个（原始）对象，那么它将导致{{jsxref("TypeError")}}。在 ES2015 中，非对象参数将被视为已被密封的普通对象，会直接返回它。
+在 ES5 中，如果该方法的参数不是一个（原始的）对象，它将会导致一个 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将按原样返回，不会有任何错误，因为根据定义，基本类型已经是不可变的。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:90]:
```suggestion
在 ES5 中，如果该方法的参数不是一个对象（即基本类型），它将会抛出一个 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将按原样返回，不会有任何错误，因为根据定义，基本类型已经是不可变的。
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 5]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
 
 被密封的对象。
 
 ## 描述
 
-通常，一个对象是{{jsxref("Object.isExtensible()", "可扩展的", "", 1)}}（可以添加新的属性）。密封一个对象会让这个对象变的不能添加新属性，且所有已有属性会变的不可配置。属性不可配置的效果就是属性变的不可删除，以及一个数据属性不能被重新定义成为访问器属性，或者反之。但属性的值仍然可以修改。尝试删除一个密封对象的属性或者将某个密封对象的属性从数据属性转换成访问器属性，结果会静默失败或抛出{{jsxref("TypeError")}}（在{{jsxref("Strict_mode", "严格模式", "", 1)}} 中最常见的，但不唯一）。
+密封一个对象等价于[阻止其扩展](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，然后将现有的[属性描述符](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#描述)更改为 `configurable: false`。这会使对象上的属性集固定。令对象的所有属性不可配置且可以防止它们从数据属性转换成访问器属性，反之亦然，但它不会防止数据属性的值被更改。尝试删除或者向密封对象添加属性，或者将数据属性和访问器属性互相转换，都将失败，可能会静默失败，也可能会抛出 {{jsxref("TypeError")}}（最常见的是在{{jsxref("Strict_mode", "严格模式", "", 1)}}代码中，但不仅限于此）。
 
-不会影响从原型链上继承的属性。但 [`Object.prototype.__proto__`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/proto) ( {{deprecated_inline}} ) 属性的值也会不能修改。
+原型链保持原样。然而，由于[扩展被阻止](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，`[[Prototype]]` 不能重新分配。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:31]:
```suggestion
原型链保持原样。然而，由于[扩展被阻止](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions)，`[[Prototype]]` 不能重新赋值。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 6]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
 
-返回被密封对象的引用。
+不同于 {{jsxref("Object.freeze()")}} 的是，通过 `Object.seal()` 密封的对象可以更改其现有属性，只要它们是可写的即可。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:33]: "即可" has the same meaning with "只要"

```suggestion
不同于 {{jsxref("Object.freeze()")}} 的是，通过 `Object.seal()` 密封的对象可以更改其现有属性，只要它们是可写的。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 7]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
-// 添加属性将会失败
+// 现在，除了属性值之外的任何更改都将失败。
+obj.quaxxor = "the friendly duck";
+// 静默不添加属性
 delete obj.foo;
-// 删除属性将会失败
+// 静默不添删除属性
 
-// 在严格模式下，这样的尝试将会抛出错误
+// ... 和严格模式下，这种尝试将会抛出 TypeErrors。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:71]:
```suggestion
// ...且严格模式下，这种尝试将会抛出 TypeError。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 8]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
-// 反之亦然
-Object.defineProperty(obj, 'foo', {
-  get: function() { return 'g'; }
-}); // throws a TypeError
+// 但不能将数据属性转换成访问者属性，反之亦然。
+Object.defineProperty(obj, "foo", {
+  get() {
+    return "g";
+  },
+}); // 抛出一个 TypeError
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:63]:
```suggestion
}); // 抛出 TypeError
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 9]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
 fail();
 
-// 通过 Object.defineProperty 添加属性将会报错
-Object.defineProperty(obj, 'ohai', {
-  value: 17
-}); // throws a TypeError
-Object.defineProperty(obj, 'foo', {
-  value: 'eit'
-}); // 通过 Object.defineProperty 修改属性值
+// 尝试通过 Object.defineProperty 进行添加也会引发。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:79]:
```suggestion
// 尝试通过 Object.defineProperty 添加属性也会抛出错误。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 10]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
+Object.defineProperty(obj, "ohai", {
+  value: 17,
+}); // 抛出一个 TypeError
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:82]:
```suggestion
}); // 抛出 TypeError
```

File: files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md [hunk 11]
```diff
@@ -5,88 +5,89 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/seal
...
+Object.defineProperty(obj, "foo", {
+  value: "eit",
+}); // 更改现有属性值
 ```
 
-## 注意
+### 非对象参数
 
-在 ES5 中，如果这个方法的参数不是一个（原始）对象，那么它将导致{{jsxref("TypeError")}}。在 ES2015 中，非对象参数将被视为已被密封的普通对象，会直接返回它。
+在 ES5 中，如果该方法的参数不是一个对象（即基本类型），它将会抛出一个 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将按原样返回，不会有任何错误，因为根据定义，基本类型已经是不可变的。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/seal/index.md:90]:
```suggestion
在 ES5 中，如果该方法的参数不是一个对象（即基本类型），它将会抛出 {{jsxref("TypeError")}}。在 ES2015 中，非对象参数将按原样返回，不会有任何错误，因为根据定义，基本类型已经是不可变的。
```

**Review by yin1999 — COMMENTED** [review]
some suggestions

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**